pkg/querier/distributor_queryable_test.go

```diff
@@ -261,6 +262,93 @@ func TestIngesterStreamingMixedResults(t *testing.T) {
 	require.NoError(t, seriesSet.Err())
 }
 
+func genTestHistogram(timestamp int64, value int) mimirpb.Histogram {
+	return mimirpb.FromHistogramToHistogramProto(timestamp, test.GenerateTestHistogram(value))
+}
+
+func genTestFloatHistogram(timestamp int64, value int) mimirpb.Histogram {
+	return mimirpb.FromFloatHistogramToHistogramProto(timestamp, test.GenerateTestFloatHistogram(value))
+}
+
+func TestIngesterStreamingMixedResultsHistograms(t *testing.T) {
+	// TODO(histograms): Add similar test for mixed samples and histograms?
```

Comment:
Yes, except the current verifySeries is using generics and not `interface{}` as the embedded type, which makes it hard to describe the expected result in one step.

Reply:
Discussed in slack, we'll make this change in a separate PR later. Issue: https://github.com/grafana/mimir/issues/4405

Reply:
```suggestion
	// Add similar test for mixed float values and histograms: https://github.com/grafana/mimir/issues/4405
```